internal/website/gatherexamples/gatherexamples.go

```diff
@@ -0,0 +1,301 @@
+					// Does not contain the signifier comment. Skip it.
+					continue
+				}
+				exampleCode := rewriteBlock(original)
+				if len(usedPackages) > 0 {
+					exampleCode = formatImports(usedPackages) + "\n\n" + exampleCode
+				}
+
+				pkgPath := strings.TrimSuffix(pkg.PkgPath, "_test")
+				exampleName := pkgPath + "." + fn.Name.Name
+				examples[exampleName] = exampleCode
+			}
+		}
+	}
+	return examples
+}
+
+// rewriteBlock reformats a Go block statement for display as an example.
+func rewriteBlock(block string) string {
+	// Trim block.
+	block = strings.TrimPrefix(block, "{")
+	block = strings.TrimSuffix(block, "}")
+
+	// Rewrite line-by-line.
+	sb := new(strings.Builder)
+rewrite:
+	for len(block) > 0 {
+		var line string
+		line, block = nextLine(block)
+
+		// Dedent line.
+		// TODO(light): In the case of a multi-line raw string literal,
+		// this can produce incorrect rewrites.
+		line = strings.TrimPrefix(line, "\t")
+
+		// Check if this is line that needs rewriting.
+		start := strings.IndexFunc(line, func(r rune) bool { return r != ' ' && r != '\t' })
+		if start == -1 {
+			// Blank.
+			sb.WriteString(line)
+			sb.WriteByte('\n')
+			continue
+		}
+		indent := line[:start]
+		switch line[start:] {
+		case "// Variables set up elsewhere:":
+			// Skip lines until we hit a blank line.
+			for len(block) > 0 {
+				var next string
+				next, block = nextLine(block)
+				if strings.TrimSpace(next) == "" {
+					break
+				}
+			}
+		case "// Ignore unused variables in example:":
+			// Ignore remaining lines.
+			break rewrite
+		case "log.Fatal(err)":
+			sb.WriteString(indent)
+			sb.WriteString("return err")
+			sb.WriteByte('\n')
+		default:
+			if !strings.HasPrefix(line[start:], signifierCommentPrefix) {
+				// Ordinary line.
+				sb.WriteString(line)
+				sb.WriteByte('\n')
+			}
+		}
+	}
+	return strings.TrimSpace(sb.String())
+}
+
+// nextLine splits the string at the next linefeed.
+func nextLine(s string) (line, tail string) {
+	i := strings.IndexByte(s, '\n')
+	if i == -1 {
+		return s, ""
+	}
+	return s[:i], s[i+1:]
+}
+
+// formatImports formats a map of imports to their package identifiers into a
+// Go import declaration.
+func formatImports(usedPackages map[string]string) string {
+	if len(usedPackages) == 0 {
+		return ""
+	}
+	if len(usedPackages) == 1 {
+		// Special case: one-line import.
+		for path, id := range usedPackages {
+			if id != "" {
+				return fmt.Sprintf("import %s %q", id, path)
+			}
+			return fmt.Sprintf("import %q", path)
+		}
+	}
+	// Typical case: multiple imports in factored declaration form.
+	// Group into standard library imports then third-party imports.
+	sortedStdlib := make([]string, 0, len(usedPackages))
+	sortedThirdParty := make([]string, 0, len(usedPackages))
+	for path := range usedPackages {
+		if strings.ContainsRune(path, '.') {
+			// Third-party imports almost always contain a dot for a domain name,
```

Comment:
Is it almost or always? Is this the logic used by goimports?

Reply:
`goimports` has a giant table of all the stdlib symbols: https://github.com/golang/tools/blob/master/imports/zstdlib.go

This heuristic works for any Go modules/GOPATH setup I can imagine. Paths like `google3/foo/bar` would be ones that break. Since this tool is only being used to generate our examples, which will be operating in Go modules only, then this heuristic is 100% accurate.